code/modules/mob/living/simple_animal/hostile/pulse_demon/pulsedemon.dm

```diff
@@ -290,6 +324,11 @@
 		visible_message("<span class ='notice'>[user] attempted to taste \the [src], for no particular reason, and got rightfully burned.</span>")
 		shockMob(user)
 		
+/mob/living/simple_animal/hostile/pulse_demon/hitby(thrown_thing)
+	visible_message("<span class ='notice'>\the [thrown_thing] goes right through \the [src]!</span>")
+	//no damage from being hit by thrown stuff but the whack sound still plays for some reason
```

Comment:
The sound plays in throw_impact not in hitby. You should prevent it from impacting the pulsedemon at all. This one is actually very easy to fix if you know the throw code. Move this to PreImpact! Returning TRUE there will make nothing thrown hit him.
```
/mob/living/simple_animal/hostile/pulse_demon/PreImpact(atom/movable/A, speed)
  return TRUE
```
I recommend no feedback message because they can see it going through him, and if 50 items get thrown at him, like from a disposal, it will spam chat badly.